### PR TITLE
new panos module for profile groups

### PR DESCRIPTION
**Pull request by ivanbojer**
##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
panos_pg

##### ANSIBLE VERSION

```
ansible 2.1.0.0 
config file = 
configured module search path = Default w/o overrides
```

##### SUMMARY
New panos module that allows for creation of security profile groups.

### CODE REVIEW

File: lib/ansible/modules/network/panos/panos_pg.py [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+        vulnerability=dict(),
+        wildfire=dict(),
+        commit=dict(type='bool', default=True)
+    )
+    module = AnsibleModule(argument_spec=argument_spec, supports_check_mode=False)
+    if not HAS_LIB:
+        module.fail_json(msg='pan-python is required for this module')
+
+    ip_address = module.params["ip_address"]
+    if not ip_address:
```
Comment (privateip) [inline lib/ansible/modules/network/panos/panos_pg.py]: not needed.  setting `required=True` in the argument_spec will automatically generate an failed message if the argument is not specified.

File: lib/ansible/modules/network/panos/panos_pg.py [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+        module.fail_json(msg="ip_address should be specified")
+    password = module.params["password"]
+    if not password:
```
Comment (privateip) [inline lib/ansible/modules/network/panos/panos_pg.py]: same as with`ip_address`

File: lib/ansible/modules/network/panos/panos_pg.py [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+    username = module.params['username']
+
+    xapi = pan.xapi.PanXapi(
+        hostname=ip_address,
+        api_username=username,
+        api_password=password
+    )
+
+    pg_name = module.params['pg_name']
+    if not pg_name:
```
Comment (privateip) [inline lib/ansible/modules/network/panos/panos_pg.py]: same as `ip_address`

File: lib/ansible/modules/network/panos/panos_pg.py [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+    commit = module.params['commit']
+
+    try:
+        changed = add_pg(xapi, pg_name, data_filtering, file_blocking,
+                         spyware, url_filtering, virus, vulnerability, wildfire)
+
+        if changed and commit:
+            xapi.commit(cmd="<commit></commit>", sync=True, interval=1)
+    except PanXapiError:
+        import sys
```
Comment (privateip) [inline lib/ansible/modules/network/panos/panos_pg.py]: use the `get_exception` function in ansible.module_utils.basic instead

```
from ansible.module_utils.basic import get_exception

.....

except PanXapiError:
   exc = get_exception()
   module.fail_json(msg=exc.message)
```

File: lib/ansible/modules/network/panos/panos_pg.py [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+    ip_address: "192.168.1.1"
+    password: "admin"
+    username: "admin"
+    pg_name: "pg-default"
+    virus: "default"
+    spyware: "default"
+    vulnerability: "default"
+'''
+
+RETURN = '''
```
Comment (privateip) [inline lib/ansible/modules/network/panos/panos_pg.py]: doesn't match actual return string

**Review by privateip — CHANGES_REQUESTED** [review]
Consider pulling common functions across all panos_* modules into a shared lib in ansible/module_utils.  Then you can import in each module using standard Python imports.

**Comment by ivanbojer** [comment]
Thank you for the review. I've incorporated all the changes as request and will follow these in the further modules. The one thing I was not sure about was RETURN documentation and I followed what was done in 'wakeonlan.py' given that this module has no return variables.

File: lib/ansible/modules/network/panos/panos_pg.py (hunk 5 above)
Comment (ivanbojer) [reply lib/ansible/modules/network/panos/panos_pg.py]: ok, I followed what was done in 'wakeonlan.py' given that this module has no return variables either.

File: lib/ansible/modules/network/panos/panos_pg.py (hunk 1 above)
Comment (ivanbojer) [reply lib/ansible/modules/network/panos/panos_pg.py]: OK

File: lib/ansible/modules/network/panos/panos_pg.py (hunk 2 above)
Comment (ivanbojer) [reply lib/ansible/modules/network/panos/panos_pg.py]: OK

File: lib/ansible/modules/network/panos/panos_pg.py (hunk 3 above)
Comment (ivanbojer) [reply lib/ansible/modules/network/panos/panos_pg.py]: OK

File: lib/ansible/modules/network/panos/panos_pg.py (hunk 4 above)
Comment (ivanbojer) [reply lib/ansible/modules/network/panos/panos_pg.py]: OK

**Comment by ivanbojer** [comment]
ready_for_review

**Pull request merged**